pootle/apps/import_export/utils.py

```diff
@@ -12,35 +12,32 @@
 from django.utils.translation import ugettext as _
 
 from pootle_store.models import Store
+from pootle_statistics.models import SubmissionTypes
 
+from .exceptions import (ERR_UNSUPPORTED_FILETYPE, ERR_MISSING_POOTLE_PATH,
+                         ERR_MISSING_POOTLE_REV, ERR_FILE_IMPORT,
+                         UnsupportedFiletypeError, MissingPootlePathError,
+                         MissingPootleRevError, FileImportError)
 
-def import_file(file):
+
+def import_file(file, user=None):
     f = getclass(file)(file.read())
+    if not hasattr(f, "parseheader"):
+        raise UnsupportedFiletypeError(_(ERR_UNSUPPORTED_FILETYPE) % file.name)
     header = f.parseheader()
     pootle_path = header.get("X-Pootle-Path")
     if not pootle_path:
-        raise ValueError(_("File '%s' missing X-Pootle-Path header\n") % (file.name))
+        raise MissingPootlePathError(_(ERR_MISSING_POOTLE_PATH) % file.name)
```

Comment:
This doesn't work as already outlined in another PR. Note the actual error messages belong here, i.e. you want to write the actual message wherever you raise the exceptions.

I see you have added these to check for exception messages in tests. However, since you already added specialized exceptions, I believe we are already covered.